NPE

```aya
def Nat => Pi (P : Prop) (P -> P) P -> P

def zero : Nat => \P s z => z
def suc (x : Nat) : Nat => \P s z => s (x P s z)

def add (x : Nat) (y : Nat) : Nat => x Nat (suc y) y
```

```
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Boolean.booleanValue()" because the return value of "org.aya.tyck.unify.TermComparator.checkParam(org.aya.core.term.Term$Param, org.aya.core.term.Term$Param, org.aya.core.visitor.Subst, org.aya.core.visitor.Subst, org.aya.tyck.unify.TermComparator$Sub, org.aya.tyck.unify.TermComparator$Sub, java.util.function.Supplier, java.util.function.BiFunction)" is null
	at org.aya.tyck.unify.TermComparator.doCompareType(TermComparator.java:410)
	at org.aya.tyck.unify.TermComparator.doCompareUntyped(TermComparator.java:423)
	at org.aya.tyck.unify.TermComparator.compareUntyped(TermComparator.java:157)
	at org.aya.tyck.unify.TermComparator.compare(TermComparator.java:138)
	at org.aya.tyck.unify.TermComparator.compare(TermComparator.java:118)
	at org.aya.tyck.tycker.UnifiedTycker.unifyTy(UnifiedTycker.java:188)
	at org.aya.tyck.tycker.UnifiedTycker.unifyTyReported(UnifiedTycker.java:75)
	at org.aya.tyck.tycker.UnifiedTycker.unifyTyReported(UnifiedTycker.java:60)
	at org.aya.tyck.tycker.UnifiedTycker.inheritFallbackUnify(UnifiedTycker.java:108)
	at org.aya.tyck.ExprTycker.doInherit(ExprTycker.java:523)
	at org.aya.tyck.ExprTycker.lambda$inherit$20(ExprTycker.java:588)
	at org.aya.tyck.tycker.ConcreteAwareTycker.traced(ConcreteAwareTycker.java:65)
	at org.aya.tyck.ExprTycker.inherit(ExprTycker.java:583)
	at org.aya.tyck.ExprTycker.doSynthesize(ExprTycker.java:280)
	at org.aya.tyck.tycker.ConcreteAwareTycker.traced(ConcreteAwareTycker.java:65)
	at org.aya.tyck.ExprTycker.synthesize(ExprTycker.java:593)
	at org.aya.tyck.ExprTycker.doSynthesize(ExprTycker.java:241)
	at org.aya.tyck.tycker.ConcreteAwareTycker.traced(ConcreteAwareTycker.java:65)
	at org.aya.tyck.ExprTycker.synthesize(ExprTycker.java:593)
	at org.aya.tyck.ExprTycker.doInherit(ExprTycker.java:523)
	at org.aya.tyck.ExprTycker.lambda$inherit$20(ExprTycker.java:588)
```

--- a/base/src/main/java/org/aya/tyck/unify/TermComparator.java
+++ b/base/src/main/java/org/aya/tyck/unify/TermComparator.java
@@ -407,7 +407,7 @@ public sealed abstract class TermComparator extends MockTycker permits Unifier {
         yield visitArgs(lhs.args(), rhs.args(), lr, rl, Term.Param.subst(Def.defTele(lhs.ref()), lhs.ulift()));
       }
       case Pair(PiTerm(var lParam, var lBody), PiTerm(var rParam, var rBody)) ->
-        checkParam(lParam, rParam, new Subst(), new Subst(), lr, rl, () -> null,
+        checkParam(lParam, rParam, new Subst(), new Subst(), lr, rl, () -> false,
           (lsub, rsub) -> compare(lBody.subst(lsub), rBody.subst(rsub), lr, rl, null));
       case Pair(SigmaTerm(var lParams), SigmaTerm(var rParams)) -> checkParams(lParams.view(), rParams.view(),
         new Subst(), new Subst(), lr, rl, () -> false, (lsub, rsub) -> true);

--- a/base/src/test/resources/success/common/src/Arith/Nat/Encoding.aya
+++ b/base/src/test/resources/success/common/src/Arith/Nat/Encoding.aya
@@ -1,0 +1,16 @@
+def Natλ => Pi (P : Prop) (P -> P) P -> P
+
+def zero : Natλ => \P s z => z
+def suc (x : Natλ) : Natλ => \P s z => s (x P s z)
+
+def add (x : Natλ) (y : Natλ) : Natλ => x Natλ suc y
+def mul (x : Natλ) (y : Natλ) : Natλ => x Natλ (add y) zero
+
+def three => suc (suc (suc zero))
+
+// Has to be in Prop, otherwise cannot fit in Natλ motive
+open data Nat : Prop
+| zero'
+| suc' Nat
+
+def toNat : Natλ -> Nat => \x => x Nat suc' 0

--- a/base/src/test/resources/success/common/src/Arith/Nat/Encoding.aya
+++ b/base/src/test/resources/success/common/src/Arith/Nat/Encoding.aya
@@ -9,8 +9,8 @@ def mul (x : Natλ) (y : Natλ) : Natλ => x Natλ (add y) zero
 def three => suc (suc (suc zero))
 
 // Has to be in Prop, otherwise cannot fit in Natλ motive
-open data Nat : Prop
+example open data Nat : Prop
 | zero'
 | suc' Nat
 
-def toNat : Natλ -> Nat => \x => x Nat suc' 0
+example def toNat : Natλ -> Nat => \x => x Nat suc' zero'

--- a/base/src/test/resources/success/src/Impredicative.aya
+++ b/base/src/test/resources/success/src/Impredicative.aya
@@ -2,3 +2,5 @@ def Nat => Pi (P : Prop) (P -> P) P -> P
 
 def zero : Nat => \P s z => z
 def suc (x : Nat) : Nat => \P s z => s (x P s z)
+
+def add (x : Nat) (y : Nat) : Nat => x Nat (suc y) y

--- a/base/src/test/resources/success/src/Impredicative.aya
+++ b/base/src/test/resources/success/src/Impredicative.aya
@@ -1,6 +1,0 @@
-def Nat => Pi (P : Prop) (P -> P) P -> P
-
-def zero : Nat => \P s z => z
-def suc (x : Nat) : Nat => \P s z => s (x P s z)
-
-def add (x : Nat) (y : Nat) : Nat => x Nat (suc y) y

--- a/ide/src/main/java/org/aya/ide/action/InlayHints.java
+++ b/ide/src/main/java/org/aya/ide/action/InlayHints.java
@@ -1,4 +1,4 @@
-// Copyright (c) 2020-2022 Tesla (Yinsen) Zhang.
+// Copyright (c) 2020-2023 Tesla (Yinsen) Zhang.
 // Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
 package org.aya.ide.action;
 
@@ -10,8 +10,8 @@ import org.aya.core.term.Term;
 import org.aya.ide.syntax.SyntaxNodeAction;
 import org.aya.ide.util.XYXY;
 import org.aya.pretty.doc.Doc;
-import org.aya.util.prettier.PrettierOptions;
 import org.aya.util.error.SourcePos;
+import org.aya.util.prettier.PrettierOptions;
 import org.jetbrains.annotations.NotNull;
 
 public record InlayHints(
@@ -29,7 +29,7 @@ public record InlayHints(
 
   @Override public @NotNull Pattern pre(@NotNull Pattern pattern) {
     if (pattern instanceof Pattern.Bind bind && bind.type().get() instanceof Term term) {
-      var type = term.toDoc(options);
+      var type = Doc.sep(Doc.symbol(":"), term.toDoc(options));
       hints.append(new Hint(bind.sourcePos(), type, true));
     }
     return Ranged.super.pre(pattern);

